crates/ra_hir/src/ty/tests/data/array.txt

```diff
@@ -1,38 +1,46 @@
 [9; 10) 'x': &str
 [18; 19) 'y': isize
-[28; 258) '{     ..."]]; }': ()
-[38; 39) 'a': [&str,]
-[42; 45) '[x]': [&str,]
+[28; 263) '{     ... []; }': ()
+[38; 39) 'a': [&str]
+[42; 45) '[x]': [&str]
 [43; 44) 'x': &str
-[55; 56) 'b': [[&str,], [&str,]]
-[59; 65) '[a, a]': [[&str,], [&str,]]
-[60; 61) 'a': [&str,]
-[63; 64) 'a': [&str,]
-[75; 76) 'c': [[[&str,], [&str,]], [[&str,], [&str,]]]
-[79; 85) '[b, b]': [[[&str,], [&str,]], [[&str,], [&str,]]]
-[80; 81) 'b': [[&str,], [&str,]]
-[83; 84) 'b': [[&str,], [&str,]]
-[96; 97) 'd': [isize,]
-[100; 103) '[y]': [isize,]
-[101; 102) 'y': isize
-[113; 114) 'e': [[isize,], [isize,]]
-[117; 123) '[d, d]': [[isize,], [isize,]]
-[118; 119) 'd': [isize,]
-[121; 122) 'd': [isize,]
-[133; 134) 'f': [[[isize,], [isize,]], [[isize,], [isize,]]]
-[137; 143) '[e, e]': [[[isize,], [isize,]], [[isize,], [isize,]]]
-[138; 139) 'e': [[isize,], [isize,]]
-[141; 142) 'e': [[isize,], [isize,]]
-[197; 198) 'g': [_, _]
-[201; 207) '[1, 2]': [_, _]
-[202; 203) '1': [unknown]
-[205; 206) '2': [unknown]
-[217; 218) 'h': [_, _]
-[221; 231) '["a", "b"]': [_, _]
-[222; 225) '"a"': [unknown]
-[227; 230) '"b"': [unknown]
-[241; 242) 'b': [[&str,], [_,]]
-[245; 255) '[a, ["b"]]': [[&str,], [_,]]
-[246; 247) 'a': [&str,]
-[249; 254) '["b"]': [_,]
-[250; 253) '"b"': [unknown]
+[55; 56) 'b': [[&str]]
+[59; 65) '[a, a]': [[&str]]
+[60; 61) 'a': [&str]
+[63; 64) 'a': [&str]
+[75; 76) 'c': [[[&str]]]
+[79; 85) '[b, b]': [[[&str]]]
+[80; 81) 'b': [[&str]]
+[83; 84) 'b': [[&str]]
+[95; 96) 'd': [isize]
+[99; 102) '[y]': [isize]
+[100; 101) 'y': isize
+[112; 113) 'e': [isize]
+[116; 128) '[y, 1, 2, 3]': [isize]
+[117; 118) 'y': isize
+[120; 121) '1': isize
+[123; 124) '2': isize
+[126; 127) '3': isize
+[137; 138) 'f': [[isize]]
+[141; 147) '[d, d]': [[isize]]
+[142; 143) 'd': [isize]
+[145; 146) 'd': [isize]
+[157; 158) 'g': [[isize]]
+[161; 167) '[e, e]': [[isize]]
+[162; 163) 'e': [isize]
+[165; 166) 'e': [isize]
+[177; 178) 'h': [_]
+[181; 187) '[1, 2]': [_]
+[182; 183) '1': [_]
+[185; 186) '2': [_]
+[197; 198) 'i': [_]
+[201; 211) '["a", "b"]': [_]
+[202; 205) '"a"': [_]
+[207; 210) '"b"': [_]
+[221; 222) 'b': [[&str]]
+[225; 235) '[a, ["b"]]': [[&str]]
+[226; 227) 'a': [&str]
+[229; 234) '["b"]': [[&str]]
+[230; 233) '"b"': [&str]
+[245; 246) 'x': [[u8]]
```

Comment:
I think this should be `'x': [u8]`.
But I don't know why is results above.

Reply:
It's because you're passing the expected type for the slice (which is `[u8]`) as the expected type for the elements. You need to extract the `u8` from the expected type and pass that down.